synapse/handlers/message.py

```diff
@@ -969,7 +978,16 @@ async def create_new_client_event(
             event.internal_metadata.outlier = True
             context = EventContext.for_outlier()
         else:
-            context = await self.state.compute_event_context(event)
+            old_state = None
+            # Define the state for historical messages while we know to get all of
+            # state_groups setup properly when we `compute_event_context`.
+            if builder.internal_metadata.is_historical() and full_state_ids_at_event:
+                old_state = await self.store.get_events_as_list(full_state_ids_at_event)
+
+            context = await self.state.compute_event_context(
+                event,
+                # old_state=old_state
```

Comment:
We can revert if we don't need `old_state` here (it's commented out atm) and all of the `auth_event_ids` stuff. All of this stuff since it's part of `v1` in https://github.com/matrix-org/synapse/pull/10948